### ISSUE TITLE
Upload to Domoticz

### ISSUE THREAD

**Issue by Reneke43** [solmoller/eversolar-monitor#14]
There are several upload options in the script. However, I would like to upload the data to Domoticz. For this you can use JSON to upload the data to virtual hardware.
I'm struggling how to get this done by modifying the Perl script.

**Comment by solmoller** [comment]
Sounds like a feasible route you have selected - did you get it to work?

**Comment by Reneke43** [comment]
I've been building other automation in Domoticz for my home. (Like a nice theater mode dimming the lights on movie start for example ;-))
Your comment triggered me to continue with this project.
Domoticz has several options to upload data. The script writes JSON text to /tmp/eversolar. This could also be used to modify a dummy electricity sensor by uploading the data to Domoticz with an URL like
http://domoticz_ip:domoticz_port/json.htm?type=command&param=udevice&idx=IDX&nvalue=0&svalue=POWER;ENERGY

**Comment by solmoller** [comment]
If you get it to work, please submit the code

BR

Henrik

Den 16/01/2017 10.15 PM skrev "Reneke43" <notifications@github.com>:

**Comment by Reneke43** [comment]
I got it working. I added a section Domoticz to the .ini file and the script. Also added a small instruction to the README.md. Also I edited the index.html so that it updates every 10 seconds. A bit more realtime ;-)
I've added my working eversolar folder, so there is data included from my inverter. I don't know how to wipe it, but I guess you do.
For now it only uploads realtime generated power. I will continue to work to add more data sensors to Domoticz. (Current, Voltage, Inverter temperature...)
[eversolar-monitor-code.zip](https://github.com/solmoller/eversolar-monitor/files/720300/eversolar-monitor-code.zip)

**Comment by solmoller** [comment]
Thanks for submitting the code, will you be progressing shortly? I'll definately add it to the code, it's good to have the project back. Incidentally, github offers quite substantial support for having multiple developers sharing a project, you might want to look into how it works.

**Comment by ronneman** [comment]
@Reneke43 ,

Any progress on the Domoticz integration. I am looking forward to this functionality as well.

**Comment by solmoller** [comment]
So this should be added to the ini:
`# upload to Domoticz home automation server (more information: https://www.domoticz.com)
enabled = 0
address = "x.x.x.x"                     
# Domoticz server ip. Can be localhost when running this script on the same system as this script.
port = 8080                             
# Domoticz port (8080 is default)
IDX = xxx                               
# virtual sensor IDX number. This can be found on the Domoticz hardware page.


`

And this to the pl:
`
$config->define("domoticz_enabled=s");
$config->define("domoticz_address=s");
$config->define("domoticz_port=s");
$config->define("domoticz_IDX=s")


 ##  Data to Domoticz
            ##
            ##
            ##
            ###############################################################################

            if($config->domoticz_enabled) {

                my $domoticz_address = $config->domoticz_address;
                my $domoticz_port = $config->domoticz_port;
                my $domoticz_IDX = $config->domoticz_IDX;

                my $cmd = `curl -s "http://$domoticz_address:$domoticz_port/json.htm?type=command&param=udevice&idx=$domoticz_IDX&nvalue=0&svalue=$pac;$e_today_wh" `;
		chomp($cmd);
                pmu_log("Severity 3, ".$inverters{$inverter}{"serial"}." uploading to domoticz, response: $cmd");

            }


            
`

Is that all, or did you change other places?

**Comment by solmoller** [comment]
Updated the code, can you test it, please?

**Comment by Reneke43** [comment]
I've just downloaded the code. I'll test tomorrow when I have some daylight over here :-)

**Comment by solmoller** [comment]
Great. Should I put your name in the code?


Den 3. dec. 2017 19.17 skrev "Reneke43" <notifications@github.com>:

**Comment by solmoller** [comment]
I have added your name to the code as "Reneke43", If you'd like your full name in let me know.

**Comment by Ctenberge** [comment]
I tried to add Domoticz and succeeded mostly, no errors in the eversolar section, only I get never data in the dummy device. 
Domoticz runs on another Pi and I am a bit confused here. I followed your instructions for the eversolar.ini file and do not understand enabled should be 0 (zero) where normally one would expect a 1 (one).
I tried both by the way and did'nt get any result. No power is being shown.
What is missing?

**Comment by Ctenberge** [comment]
I found it. For some reason this line was commented
 my $cmd = `curl -s "http://$domoticz_address:$domoticz_port/json.htm?type=command&param=udevice&idx=$domoticz_IDX&nvalue=0&svalue=$pac;$e_today_wh" `;

now it works.
Thanks

**Comment by solmoller** [comment]
Thanks for the fix, version 0.171 released with your name in it

**Comment by Ctenberge** [comment]
Hello Henrik,

the quotes *in*  the expression are neccesarily and must therefore be
present.
I did'nt get Domoticz at work first and tried several ways to get the job
done, and I commented the whole line.
Later on I forget about it, removed my efforts and it still did'nt work.
So I posted my problem and shortley afterwards discovered the remaining
qoute at the beginning of the line (a #).
After removing, Domoticz  started the data from eversolar.
I posted the result but failed to tell about the # , sorry.

with regards, Carel ten Berge

2018-02-03 11:59 GMT+01:00 Henrik Møller Jørgensen <notifications@github.com
